src/auto-evo/mutation_strategy/UpgradeOrganelle.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Linq;
+using AutoEvo;
+
+public class UpgradeOrganelle : IMutationStrategy<MicrobeSpecies>
+{
+    private readonly FrozenSet<OrganelleDefinition> allOrganelles;
+    private readonly IComponentSpecificUpgrades upgrade;
+
+    public UpgradeOrganelle(Func<OrganelleDefinition, bool> criteria, IComponentSpecificUpgrades upgrade)
+    {
+        allOrganelles = SimulationParameters.Instance.GetAllOrganelles().Where(criteria).ToFrozenSet();
+        this.upgrade = upgrade;
+    }
+
+    public bool Repeatable => true;
```

Comment:
Is this step repeatable BTW because this doesn't do anything different if this is called multiple times on the same species?

Reply:
honestly I had just assumed that "repeatable" just mean it wouldn't break if you ran it twice. I'll set it to false.

Reply:
That's true, but the auto-evo system will burn unnecessary CPU cycles doing a mutation again that doesn't result in any changes (if I understood the new core code correctly, any repeatable mutation modifier can be tried again on the same species once it has been already applied if there is any MP left, which there will be as your code doesn't reduce the MP count).